src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalCatchCheck.java

```diff
@@ -154,7 +155,8 @@ public final class IllegalCatchCheck extends AbstractCheck {
     /** Specify exception class names to reject. */
     private final Set<String> illegalClassNames = Arrays.stream(new String[] {"Exception", "Error",
         "RuntimeException", "Throwable", "java.lang.Error", "java.lang.Exception",
-        "java.lang.RuntimeException", "java.lang.Throwable", }).collect(Collectors.toSet());
+        "java.lang.RuntimeException", "java.lang.Throwable", })
+            .collect(Collectors.toCollection(HashSet::new));
```

Comment:
I don't use these type of methods much, so I am not really following. Can you explain what the difference or harm is from the old code?

> occurrences that do need a guarantee on mutability are updated 

Don't we want these to be immutable?

Reply:
> I don't use these type of methods much, so I am not really following. Can you explain what the difference or harm is from the old code?  

In the cases that are updated in this PR we are making some assumptions about the mutability of the sets (and lists). To be precise: we are using a collector to create a set and later modify the set, so the set is mutated. The Javadoc of the previously used method clearly states that: "There are no guarantees on the type, mutability, serializability, or thread-safety of the Set returned; if more control over the returned Set is required, use `toCollection(Supplier)`."   
    
Since we want to have more control over the returned set (because we need to have the guarantee that the set is mutable), we need to rewrite this code to the `toCollection(Supplier)`.  

There are some [extra examples on the documentation](https://error-prone.picnic.tech/bugpatterns/CollectorMutability/) of this check for some examples and its suggested fixes.

Reply:
Thanks for the explanation.

> There are no guarantees on ... mutability, 

Now I am just curious. What causes it to change mutability for the old method call? We have not had any issues with this code and I am sure it has been around for months. Is this just JRE implementation specific?

Reply:
Hmm, I'm not sure I can answer that 🤔. I'd suggest looking at the actual [implementation](https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/util/stream/Collectors.java#L289) of the method, maybe that explains it?

Reply:
Alright. ;)
Thanks anyways.

Reply:
Even in most cases such updates looks like more verbose but more exact.
the  most obvious reason to be verbose is to at least demand engineer to decide on mutable vs immutable. That part is good.

In our project we do not benefit much with different types of Sets, Lists, so in most cases it will be default HashSet, ArrayList, HashMap. But think it is still good to think on type of collection.